subprojects/base-services/src/main/java/org/gradle/internal/operations/BuildOperationWorker.java

```diff
@@ -16,16 +16,16 @@
 
 package org.gradle.internal.operations;
 
-import org.gradle.api.Action;
-
 /**
- * A worker that can run build operations of type {@link T}.
+ * A worker that can run build operations.
  *
  * Implementations must be thread-safe.
  */
-public interface BuildOperationWorker<T extends BuildOperation> extends Action<T> {
+public interface BuildOperationWorker<O extends BuildOperation> {
     /**
      * Returns a human consumable name for this tool.
      */
     String getDisplayName();
+
+    void execute(O buildOperation);
```

Comment:
That's a newly introduced bug on this branch and we need to fix it. The worker needs to be wrapped so that it executes through the same code path as everything else. See the [behavior](https://github.com/gradle/gradle/blob/3f063a2fc9c9d4d230008b846fa244a6520ff5b9/subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationProcessor.java#L110-L110) on master.

This could mean we're missing test coverage.